Advanced-Recruit-Training/src/BatteryStateOfChargeService.cpp

```diff
@@ -16,22 +25,78 @@ BatteryStateOfChargeService::~BatteryStateOfChargeService()
 
 double BatteryStateOfChargeService::totalAmpHoursUsed() const
 {
-    return 0.0;
+
+
+     return amphoursUsed;
 }
 
 bool BatteryStateOfChargeService::isCharging() const
 {
-    return false;
+if(averageCurrent<0){
+    return true;
+}else
+{return false;}
+
 }
 
 QTime BatteryStateOfChargeService::timeWhenChargedOrDepleted() const
 {
-    return QTime::currentTime();
+
+    double timeUntilCharged;
+    double timeUntilDepletion;
+
+   QTime charged  (0,0,0,0);
+   timeUntilCharged= (amphoursUsed/averageCurrent)*3600*1000;
+
+   charged = charged.addMSecs(timeUntilCharged);
+
+
+
+   QTime depletion (0,0,0,0);
+   timeUntilDepletion=((BATTERY_AMP_HOUR_CAPACITY-amphoursUsed)*3600000)/averageCurrent;
+   depletion = depletion.addMSecs(timeUntilDepletion);
+
+   if (isCharging()==true){
+        return charged;
+    }else{
+        return depletion;
+    }
+
+
 }
 
 void BatteryStateOfChargeService::addData(const BatteryData& batteryData)
 {
-    Q_UNUSED(batteryData);
+
+
+
+
+
+
+
+
+     double initialAmountOfAmphours = (initialStateOfChargePercent_/100) * BATTERY_AMP_HOUR_CAPACITY;
+
+    if(check1)
+    {
+    averageCurrent = (batteryData.current+previousCurrent)/2;
```

Comment:
Say the program is reading the 3rd input and the last 2 currents were -1A and 5A, then (-1A + 5A + batteryData.current) / 2 does not give the average current